streams/src/test/java/org/apache/kafka/test/StreamsTestUtils.java

```diff
@@ -79,6 +83,27 @@ public static Properties getStreamsConfig() {
         return getStreamsConfig(UUID.randomUUID().toString());
     }
 
+    public static void startKafkaStreamsAndWaitForRunningState(final KafkaStreams kafkaStreams) {
+        startKafkaStreamsAndWaitForRunningState(kafkaStreams, DEFAULT_MAX_WAIT_MS);
+    }
+
+    public static void startKafkaStreamsAndWaitForRunningState(final KafkaStreams kafkaStreams,
+                                                               final long timeoutMs) {
+        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        kafkaStreams.setStateListener((newState, oldState) -> {
+            if (newState == KafkaStreams.State.RUNNING) {
+                countDownLatch.countDown();
+            }
+        });
+
+        kafkaStreams.start();
+        try {
+            countDownLatch.await(timeoutMs, TimeUnit.MILLISECONDS);
+        } catch (final InterruptedException interruptedException) {
+            throw new AssertionError("KafkaStreams did not transit to RUNNING state within " + timeoutMs + " milli seconds.");
```

Comment:
The error message is not accurate because it indicates that the wait was for timeoutMs, but it was interrupted so it is quite likely that the wait was less. As we don't expect an interrupt anyway rethrowing it seems the easiest way to handle this?

Reply:
+1